File: problems/arrays/largest-element-in-array/largest_element.cpp
// largest_element.cpp
#include <iostream>
#include <vector>
#include <stdexcept>
using namespace std;

int largestElement(const vector<int>& arr) {
    if (arr.empty()) {
        throw invalid_argument("Array is empty.");
    }

    int maxElement = arr[0];
    for (size_t i = 1; i < arr.size(); i++) {
        if (arr[i] > maxElement) {
            maxElement = arr[i];
        }
    }
    return maxElement;
}

int main() {
    cout << "Running test cases for C++ solution...\n\n";

    // Test 1
    vector<int> arr1 = {3, 5, 7, 2, 8};
    cout << "Input: [3, 5, 7, 2, 8]\n";
    cout << "Largest Element: " << largestElement(arr1) << "\n\n";

    // Test 2
    vector<int> arr2 = {-5, -2, -10};
    cout << "Input: [-5, -2, -10]\n";
    cout << "Largest Element: " << largestElement(arr2) << "\n\n";

    // Test 3
    vector<int> arr3 = {1, 1, 1};
    cout << "Input: [1, 1, 1]\n";
    cout << "Largest Element: " << largestElement(arr3) << "\n";

    return 0;
}

File: problems/arrays/largest-element-in-array/largest_element.py
# largest_element.py

def largest_element(arr):
    if not arr:
        raise ValueError("Array is empty.")
    max_element = arr[0]
    for num in arr[1:]:
        if num > max_element:
            max_element = num
    return max_element


#  Test cases
if __name__ == "__main__":
    print("Running test cases for Python solution...\n")

    # Test 1
    arr1 = [3, 5, 7, 2, 8]
    print("Input:", arr1)
    print("Largest Element:", largest_element(arr1))
    print()

    # Test 2
    arr2 = [-5, -2, -10]
    print("Input:", arr2)
    print("Largest Element:", largest_element(arr2))
    print()

    # Test 3
    arr3 = [1, 1, 1]
    print("Input:", arr3)
    print("Largest Element:", largest_element(arr3))
